fastplotlib/widgets/image.py

```diff
@@ -855,6 +856,30 @@ def reset_vmin_vmax(self):
 
             self.vmin_vmax_sliders[i].set_state(state)
 
+    def set_data(self, new_data: Union[np.ndarray, List[np.ndarray]], reset_vmin_vmax: bool = True, reset_current_index: bool = True):
+        """Change data of widget"""
+        if reset_current_index:
+            self.current_index["t"] = 0
+            self.sliders["t"].value = 0
+
+        if isinstance(new_data, np.ndarray) and isinstance(self.plot, Plot):
+            if new_data.shape != self._data[0].shape:
+                raise ValueError(f"new data shape {new_data.shape} does not equal current data shape {self._data[0].shape}")
+            self._data[0] = new_data
+            self.plot.graphics[0].data = self._data[0][self.current_index["t"]]
+        else:
+            if len(self._data) != len(new_data):
+                raise ValueError(f"number of new data arrays {len(new_data)} must match current number of data arrays {len(self._data)}")
+            positions = list(product(range(self.plot.shape[0]), range(self.plot.shape[1])))
+            for i, (new_darray, darray) in enumerate(zip(new_data, self._data)):
+                if new_darray.shape != darray.shape:
+                    raise ValueError(f"new data shape {new_darray.shape} does not equal current data shape {darray.shape}")
+                self._data[i] = new_data[i]
+                self.plot[positions[i][0], positions[i][1]].graphics[0].data = self._data[i][self.current_index["t"]]
```

Comment:
add a comment that we assume `graphics[0]` is always the "image widget manged data"